sklearn/linear_model/tests/test_coordinate_descent.py

```diff
@@ -302,7 +303,33 @@ def test_lasso_cv_positive_constraint():
     assert min(clf_constrained.coef_) >= 0
 
 
-# FIXME: 'normalize' to be removed in 1.2
+def _scale_alpha_inplace(estimator, n_samples):
+    """Rescale the parameter alpha from when the estimator is evoked with
+    normalize set to True to when it is evoked in a Pipeline with normalize set
+    to False and with a StandardScaler.
+    """
+    if 'alpha' not in estimator.get_params():
+        return
+
+    if isinstance(estimator, (Lasso, LassoLars, MultiTaskLasso)):
+        alpha = estimator.alpha * np.sqrt(n_samples)
+    if isinstance(estimator, (Ridge, RidgeClassifier)):
+        alpha = estimator.alpha * n_samples
+    if isinstance(estimator, (ElasticNet, MultiTaskElasticNet)):
+        if estimator.l1_ratio == 1:
+            alpha = estimator.alpha * np.sqrt(n_samples)
+        elif estimator.l1_ratio == 0:
+            alpha = estimator.alpha * n_samples
+        else:
+            # To avoid silent errors in case of refactoring
+            raise NotImplementedError
+
+    estimator.set_params(alpha=alpha)
+
+
+# FIXME: 'normalize' to be removed in 1.2 for all the models excluding:
+# OrthogonalMatchingPursuit, Lars, LassoLars, LarsCV, LassoLarsCV
```

Comment:
@agramfort @jeremiedbb I wonder what impact this will have on sparse coding in `(MiniBatch)DictionaryLearning`: I assume that for sparse coding we might not necessarily want to refit a `StandardScaler` each time to preserve the old behavior. I am not sure if we are suppose to normalize/standardize the inputs or not in a sparse coding context.

Reply:
for sparse coding you do not normalize / standardize the dictionary. It's already scaled when it's updated.

Reply:
In dictionary learning, Lasso and Lars is fitted with normalize=False and fit_intercept=False

Reply:
ok good, I was afraid we would have had to deal with complex deprecation handling.